R/LinkedMsExperiment.R

```diff
@@ -0,0 +1,366 @@
+#' @include MsExperiment.R
+
+#' @title Mass spectrometry experiments with explicit links between data entities
+#'
+#' @name LinkedMsExperiment
+#'
+#' @description
+#'
+#' The `linkSampleData` function allows to add or define *explicit* links
+#' between samples (i.e. rows in the `sampleData` `DataFrame`) and other data
+#' elements in the `MsExperiment` object. Links can be added between samples
+#' and spectra (in the `@spectra` slot) but also between individual samples and
+#' elements in any other data slot of the `MsExperiment`, such as
+#' `experimentFiles` or `otherData`. Links allow also a n:m mapping between
+#' samples and other data elements.
+#'
+#' Importantly, the presence of such links enables a (coherent) subsetting of an
+#' `MsExperiment` by samples. Thus, once the link is defined, any subsetting by
+#' sample will also correctly subset the linked data. All other, not linked,
+#' data elements are always retained as in the original `MsExperiment`.
+#'
+#'
+#' @section Accessing data within an `LinkedMsExperiment`:
+#'
+#' Naming of data elements in an `LinkedExperiment` follows the schema
+#' `"<slot name>.<element name>"` (similar to SQL
+#' `"<database table>.<column>"`). Using the shortcut
+#' `$<slot name>.<element name>` it is thus also possible to directly access
+#' data elements in the different slots of the object. Note that if a element
+#' with the specified name does not exist in the slot `NULL` is returned.
+#'
+#'
+#' @section Implementation notes:
+#'
+#' Links from samples to any other element are stored as an `integer` `matrix`
+#' as a list element of the `@sampleDataLinks` slot. The link name (i.e.
+#' the name of the list element) needs to be the name of the slot that is
+#' linked (for slots `@spectra` and `@qfeatures`) or an element within that
+#' slot (for `@experimentFiles`, `@otherData`, `@metadata`). For the latter the
+#' name needs to be in the format `<slot name>.<element name>`, e.g.
+#' `"experimentFiles.mzML_files"` to link samples to an element called
+#' `"mzML_files"` in `@experimentFiles`.
+#'
```

Comment:
Shouldn't `fromIndex` and `toIndex` be documented in here too?

Reply:
Sure, I will fix.

Reply:
documentation added.